### PR TITLE
Add models rank json file

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Add models rank json file. This file ranks all output models from different EPs. This json file includes `model_config` and `metrics`. User can find model path and EP info from `model_config`.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/engine/packaging/packaging_generator.py [hunk 1]
```diff
@@ -67,6 +69,28 @@ def _generate_zipfile_output(
...
+        itertools.chain.from_iterable(f.nodes.values() for f in footprints.values()),
+        key=lambda x: tuple(
+            x.metrics.value[metric].value if x.metrics.cmp_direction[metric] == 1 else -x.metrics.value[metric].value
+            for metric in metrics_dict
+        ),
+        reverse=True,
+    )
+    with (tempdir / "models_rank.json").open("w") as f:
+        rank = 1
+        for node in sorted_nodes:
```
Comment (guotuofeng) [inline olive/engine/packaging/packaging_generator.py:84]: is it better to write once if we put all these model info into a list?

**Review by guotuofeng — CHANGES_REQUESTED** [review]
The result json might not valid?

File: olive/engine/packaging/packaging_generator.py (hunk 1 above)
Comment (guotuofeng) [reply olive/engine/packaging/packaging_generator.py:84]: when I try to load the following json file, the following errors will be raised.
image


```json
{
    "a": "b"
}
{
    "a": "b"
}
{
    "a": "b"
}
```

Comment (xiaoyu-work) [reply olive/engine/packaging/packaging_generator.py:84]: Oh cause I don't expect user to check this file in their code, but they may want to check this file manually. So this file is not a valid json file. But you are right they may want to load this file in their code. I can update this.

File: test/unit_test/engine/packaging/test_packaging_generator.py [hunk 2]
```diff
@@ -67,6 +67,7 @@ def test_generate_zipfile_artifacts(mock_sys_getsizeof, save_as_external_data, m
         zip_ref.extractall(output_dir)
     assert (output_dir / "SampleCode").exists()
     assert (output_dir / "CandidateModels").exists()
+    assert (output_dir / "models_rank.json").exists()
```
Comment (guotuofeng) [inline test/unit_test/engine/packaging/test_packaging_generator.py:70]: better load the json to verify the format is correct?

File: olive/engine/packaging/packaging_generator.py [hunk 3]
```diff
@@ -67,6 +69,30 @@ def _generate_zipfile_output(
         shutil.move(f"{packaging_config.name}.zip", output_dir / f"{packaging_config.name}.zip")
 
 
+def _package_models_rank(tempdir, footprints: Dict["AcceleratorSpec", "Footprint"]):
+    metrics_dict = next(iter(footprints.values())).objective_dict
```
Comment (guotuofeng) [inline olive/engine/packaging/packaging_generator.py:73]: just curious the reason that we only get the first node metrics?

File: test/unit_test/engine/packaging/test_packaging_generator.py (hunk 2 above)
Comment (xiaoyu-work) [reply test/unit_test/engine/packaging/test_packaging_generator.py:70]: Added to test.

File: olive/engine/packaging/packaging_generator.py (hunk 3 above)
Comment (xiaoyu-work) [reply olive/engine/packaging/packaging_generator.py:73]: All nodes should have the same metrics dict so getting the metrics dict from any of those should be fine.

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**